src/plugins/timeline-sensor/sensor-timeline.ts

```diff
@@ -1122,201 +897,220 @@ export class SensorTimeline extends KeepTrackPlugin {
+          timeManagerInstance.calculateSimulationTime();
+          keepTrackApi.runEvent(KeepTrackApiEvents.updateDateTime, new Date(timeManagerInstance.dynamicOffsetEpoch + timeManagerInstance.staticOffset));
 
-            selectSatManagerInstance.selectSat(-1);
-            selectSatManagerInstance.selectSat(currentSatId);
-          }
-        });
+          const selectSatManagerInstance = keepTrackApi.getPlugin(SelectSatManager)!;
+          const currentSatId = selectSatManagerInstance.selectedSat;
 
+          selectSatManagerInstance.selectSat(-1);
+          selectSatManagerInstance.selectSat(currentSatId);
+        }
       });
 
-      // If no passes draw a light gray bar to indicate no passes
-      if (sensorPass.passes.length === 0) {
-        this.ctx_.fillStyle = 'rgba(200, 200, 200, 0.2)';
-        this.ctx_.fillRect(leftOffset, y - 10, width, 20);
+    });
+
+    // If no passes draw a light gray bar to indicate no passes
+    if (sensorPass.passes.length === 0) {
+      this.ctx_.fillStyle = 'rgba(200, 200, 200, 0.2)';
+      this.ctx_.fillRect(this.leftOffset, y - height / 2, this.width, height);
 
-        const drawEvent = (mouseX: number, mouseY: number): boolean => {
-          if (mouseX >= leftOffset && mouseX <= leftOffset + width && mouseY >= y - 10 && mouseY <= y + 10) {
-            const text = `${sensorPass.sensor.uiName}: No Passes`;
+      const drawEvent = (mouseX: number, mouseY: number): boolean => {
+        if (mouseX >= this.leftOffset && mouseX <= this.leftOffset + this.width && mouseY >= y - 10 && mouseY <= y + 10) {
+          const text = `${sensorPass.sensor.uiName}: No Passes`;
 
-            this.drawTooltip_(text, mouseX, mouseY);
+          this.drawTooltip_(text, mouseX, mouseY);
 
-            return true;
-          }
+          return true;
+        }
 
-          return false;
-        };
+        return false;
+      };
+
+      this.drawEvents_[`${index} - ${sensorPass.sensor.id} - no - passes`] = drawEvent;
+    }
 
-        this.drawEvents_[`${index} - ${sensorPass.sensor.id} - no - passes`] = drawEvent;
-      }
-    });
   }
 
-  private handleOnMouseMove_(event: MouseEvent): void {
-    // clear canvas
-    this.ctx_.reset();
+  private async getWeather_(lats: number[], lons: number[], start_date: Date, end_date: Date): Promise<weatherReport[]> {
 
-    // Draw static elements
-    this.ctx_.drawImage(this.canvasStatic_, 0, 0);
+    const params = {
+      "latitude": lats,
+      "longitude": lons,
+      "start_date": start_date.toISOString().slice(0, 10),
+      "end_date": end_date.toISOString().slice(0, 10),
+      "hourly": "cloud_cover",
+    };
 
-    const rect = this.canvas_.getBoundingClientRect();
-    const mouseX = event.clientX - rect.left;
-    const mouseY = event.clientY - rect.top;
+    // console.log(params);
 
-    let isHoveringOverPass = false;
+    const url = "https://api.open-meteo.com/v1/forecast";
+    const responses = await fetchWeatherApi(url, params);
```

Comment:
Need to verify this doesn't break if the external server fails to respond or gives a bad response.